src/panels/lovelace/views/hui-view.ts

```diff
@@ -169,9 +190,62 @@ export class HUIView extends ReactiveElement {
             oldLovelace.config.views[this.index]))
     ) {
       this._initializeConfig();
+      return;
+    }
+
+    if (!changedProperties.has("hass")) {
+      return;
+    }
+
+    const oldHass = changedProperties.get("hass") as HomeAssistant | undefined;
+    const viewConfig = this.lovelace.config.views[this.index];
+    if (oldHass && this.hass && this.lovelace && isStrategyView(viewConfig)) {
+      if (
+        oldHass.entities !== this.hass.entities ||
+        oldHass.devices !== this.hass.devices ||
+        oldHass.areas !== this.hass.areas ||
+        oldHass.floors !== this.hass.floors
+      ) {
+        if (this.hass.config.state === "RUNNING") {
+          // If the page is not rendered yet, we can force the refresh
+          if (this._rendered) {
+            this._debounceRefreshConfig(false);
+          } else {
+            this._refreshConfig(true);
+          }
+        }
+      }
     }
   }
 
+  private _debounceRefreshConfig = debounce(
+    (force: boolean) => this._refreshConfig(force),
+    200
+  );
+
+  private _refreshConfig = async (force: boolean) => {
+    if (!this.hass || !this.lovelace) {
+      return;
+    }
+    const viewConfig = this.lovelace.config.views[this.index];
+
+    if (!isStrategyView(viewConfig)) {
+      return;
+    }
+
+    const oldConfig = this._layoutElementConfig;
+    const newConfig = await this._generateConfig(viewConfig);
+
+    // Don't ask if the config is the same
+    if (!deepEqual(newConfig, oldConfig)) {
+      if (force) {
+        this._initializeConfig();
```

Comment:
What is the benefit of this? I think it will only help if only the view is changed and not the dashboard config? Otherwise we are doing this twice?

Reply:
This would be nice btw, if we would just re-use the config we just generated, instead of generating it again in `_initializeConfig`